docs/getting-started/basics.md

```diff
@@ -1,7 +1,150 @@
\ No newline at end of file
+The high level interactions of loading data and querying are demonstrated in the diagrams below.
+
+```mermaid
+flowchart TD
+    subgraph Graph Query
+        E((User)) --> F
+        G{{Schema}} --> F
+        F([Query]) ---> H(Gaffer)
+        J(key-value store) <--> H
+        H --> K([Result])
+    end
+    subgraph Data Input
+        A(Data)-->B{{Schema}}
+        B --> C(Gaffer)
+        C --> D(key-value store)
+    end
+```
+
+## Gaffer Schemas
+
+One of the main differences between Gaffer and other graph database tools are its schemas. In Gaffer
+JSON based schemas need to be written upfront for it to understand how to load and treat the data in
+the graph. These schemas define all aspects of the nodes and edges in the graph, and can even be
+used to automatically do basic analysis or aggregation on queries and ingested data.
+
+You can kind of think of the schema as sort of a filter or validator for the incoming data as, a
+given bit of data must conform with part of the schema or it will simply be ingored as it doesn't
+fit the structure of the graph.
+
+### Elements Schema
+
+All distinct bits of data in a Gaffer graph (e.g. nodes and edges) are refered to as 'elements'. The
+structure and properties of these graph elements are defined in the elements schema. The general
+format of an element schema are two lists; one of the `"edges"` and the other of the `"entities"`
+like the following:
+
+!!! example "Basic elements syntax"
+
+    ```json
+    {
+        "edges": {
+            "Edge": {
+                "source": "type",
+                "destination": "type",
+                "directed": "true",
+                "properties": {
+                    "property": "type"
+                }
+            }
+        },
+        "entities": {
+            "Node": {
+                "description": "A Node",
+                "vertex": "type",
+                "properties": {
+                    "property": "type"
+                }
+            }
+        }
+    }
+    ```
+
+As you can see there are a few fields for both the example `"Edge"` and `"Node"`, many of these
+require a type as their value (discussed in the [next section](#types-schema)) which, are
+essentially handlers or object types for the value associated with the field.
+
+For an `edge` the following fields are required.
+
+- `source` - A user defined type for the source node the edge originated from.
+
+- `directed` - Boolean true or false (this is still needs to be user defined in the types schema) to
+    define if the edge is directed or not. When an Edge is undirected in Gaffer, it is treated as if
+    the relationship was bidirectional and the vertices of the edge do not have an authoritative
+    source and destination.
+
+- `destination` - A user defined type for the destination node the edge goes to.
+
+For an entity only one field is actually required outlined, vertex (see below). The example does
```

Comment:
This first sentence about entities is a bit jumbled and just needs re-writing :)